### PR TITLE
[nfc] Move NFC code to the CHIP platform layer

### PR DESCRIPTION
**Pull request by Damian-Nordic**
When NFC support was added to nRF Connect examples, NFC PR
had not yet been merged to the CHIP spec. Now that NFC is
officially one of supported methods of sharing the
commissioning information, it makes sense to move the NFC
code to the platform layer of the CHIP library.

### CODE REVIEW

File: src/platform/Zephyr/NFCManagerImpl.cpp [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        return CHIP_ERROR_INTERNAL;
+    }
+
+    ChipLogProgress(DeviceLayer, "NFC Tag emulation stopped");
+    mIsStarted = false;
+    return CHIP_NO_ERROR;
+}
+
+} // namespace DeviceLayer
+} // namespace chip
```
Comment (kkasperczyk-no) [inline src/platform/Zephyr/NFCManagerImpl.cpp:106]: Missing newline

File: src/include/platform/internal/GenericPlatformManagerImpl.cpp [hunk 2]
```diff
@@ -102,6 +102,16 @@ CHIP_ERROR GenericPlatformManagerImpl<ImplClass>::_InitChipStack()
     }
     SuccessOrExit(err);
 
+    // Initialize the NFC Manager.
+#if CHIP_DEVICE_CONFIG_ENABLE_NFC
+    err = NFCMgr().Init();
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(DeviceLayer, "NFC Manager initialization failed: %s", ErrorStr(err));
```
Comment (kkasperczyk-no) [inline src/include/platform/internal/GenericPlatformManagerImpl.cpp:110]: How about `VerifyOrExit(err, ChipLogError(DeviceLayer, "NFC Manager initialization failed: %s", ErrorStr(err)))` ?

File: examples/lock-app/nrfconnect/main/AppTask.cpp [hunk 3]
```diff
@@ -418,16 +393,11 @@ void AppTask::StartBLEAdvertisementHandler(AppEvent * aEvent)
...
-    ARG_UNUSED(arg);
-    if ((event->Type == DeviceEventType::kServiceProvisioningChange) && ConnectivityMgr().IsThreadProvisioned())
+    if (event->Type == DeviceEventType::kCHIPoBLEAdvertisingChange && event->CHIPoBLEAdvertisingChange.Result == kActivity_Stopped)
     {
-        const int result = sNFC.StopTagEmulation();
-        if (result)
-        {
-            LOG_ERR("Stopping NFC Tag emulation failed");
-        }
+        NFCMgr().StopTagEmulation();
```
Comment (kkasperczyk-no) [inline examples/lock-app/nrfconnect/main/AppTask.cpp:400]: Please add checking if tag is started before trying to stop. There is a bug in the code, that when you use test mode (button 3) without previously pressing button 4, we try to stop tag that is not started and it results in error printed in logs. I pushed PR fixing it: https://github.com/project-chip/connectedhomeip/pull/5655

**Review by kkasperczyk-no — COMMENTED** [review]
Looks good overall, but please fix mentioned bug with stopping NFC tag.

File: examples/lock-app/nrfconnect/main/AppTask.cpp (hunk 3 above)
Comment (Damian-Nordic) [reply examples/lock-app/nrfconnect/main/AppTask.cpp:400]: I added the check inside `StopTagEmulation()`.

File: src/include/platform/internal/GenericPlatformManagerImpl.cpp (hunk 2 above)
Comment (Damian-Nordic) [reply src/include/platform/internal/GenericPlatformManagerImpl.cpp:110]: Yeah, I wanted to be consistent with the existing code, but can change it.

File: examples/lock-app/nrfconnect/main/AppTask.cpp (hunk 3 above)
Comment (kkasperczyk-no) [reply examples/lock-app/nrfconnect/main/AppTask.cpp:400]: Oh right, I missed it

**Review by kkasperczyk-no — APPROVED** [review]

**Pull request closed without merging**